crates/runtime/src/dataconnector/azure.rs

```diff
@@ -0,0 +1,256 @@
+    ParameterSpec::connector("use_cli")
+        .description("Set if the Azure Cli should be used for acquiring access tokens."),
+    ParameterSpec::connector("skip_signature")
+        .description("Skip fetching credentials and skip signing requests. Used for interacting with public containers."),
+    ParameterSpec::connector("disable_tagging")
+        .description("Ignore any tags provided to put_opts"),
+
+    // Common listing table parameters
+    ParameterSpec::runtime("file_format"),
+    ParameterSpec::runtime("file_extension"),
+    ParameterSpec::runtime("csv_has_header")
+        .description("Set true to indicate that the first line is a header."),
+    ParameterSpec::runtime("csv_quote").description("The quote character in a row."),
+    ParameterSpec::runtime("csv_escape").description("The escape character in a row."),
+    ParameterSpec::runtime("csv_schema_infer_max_records")
+        .description("Set a limit in terms of records to scan to infer the schema."),
+    ParameterSpec::runtime("csv_delimiter")
+        .description("The character separating values within a row."),
+    ParameterSpec::runtime("file_compression_type")
+        .description("The type of compression used on the file. Supported types are: GZIP, BZIP2, XZ, ZSTD, UNCOMPRESSED"),
+];
+
+pub const PREFIXES: [&'static str; 6] = ["azure", "abfs", "abfss", "adl", "adls", "az"];
+
+pub fn is_azure_url(url: &str) -> bool {
+    PREFIXES.iter().any(|prefix| url.starts_with(prefix))
+}
+
+impl DataConnectorFactory for AzureFactory {
+    fn create(
+        &self,
+        mut params: Parameters,
+    ) -> Pin<Box<dyn Future<Output = super::NewDataConnectorResult> + Send>> {
+        if let Some(sas_token) = params.get("sas_string").expose().ok() {
+            if sas_token.starts_with('?') {
+                let sas_token = sas_token[1..].to_string();
+                params.insert("sas_string".to_string(), sas_token.into());
+            }
+        }
+
+        Box::pin(async move {
+            let access_key = params.get("access_key").expose().ok();
+            let bearer_token = params.get("bearer_token").expose().ok();
+            let sas_string = params.get("sas_string").expose().ok();
+
+            match (access_key, bearer_token, sas_string) {
+                (Some(_), None, None)
+                | (None, Some(_), None)
+                | (None, None, Some(_))
+                | (None, None, None) => {
+                    let azure = Azure { params };
+                    Ok(Arc::new(azure) as Arc<dyn DataConnector>)
+                }
+                _ => Err(Box::new(Error::InvalidKeyAuthCombination)
+                    as Box<dyn std::error::Error + Send + Sync>),
+            }
+        })
+    }
+
+    fn prefix(&self) -> &'static str {
+        &self.prefix
+    }
+
+    // We need a unified parameter prefix for all Azure connectors
+    fn parameter_prefix(&self) -> &'static str {
+        "azure"
```

Comment:
I don't think we should have this. We should have a common data component, and multiple connectors with specific prefixes. In this case, I am thinking `azstorage` makes the most sense?